data/src/main/java/com/globant/equattrocchio/data/ImagesServicesImpl.java

```diff
@@ -29,12 +32,12 @@ public ImagesServicesImpl() {
     }
 
     @Override
-    public void getLatestImages(final Observer<String> observer) {
+    public void getLatestImages(final Observer<List<ImageEntity>> observer) {
         api.getImages().enqueue(new Callback<Result>() {
             @Override
             public void onResponse(Call<Result> call, Response<Result> response) {
                 if (response.isSuccessful()) {
-                    observer.onNext(gson.toJson(response.body()));
+                    observer.onNext(new ResultDataMapper().transform(response.body(), null));
```

Comment:
you should have a getMapper for this case. Otherwise we would have lot of instances of ResultDataMapper